src/contract.rs

```diff
@@ -45,6 +53,26 @@ pub fn instantiate(
 
     TOKEN2.save(deps.storage, &token2)?;
 
+    let owner = msg.owner.map(|h| deps.api.addr_validate(&h)).transpose()?;
+    OWNER.save(deps.storage, &owner)?;
+
+    let protocol_fee_recipient = deps.api.addr_validate(&msg.protocol_fee_recipient)?;
+    let total_fee_percent = msg.lp_fee_percent + msg.protocol_fee_percent;
```

Comment:
Decimal doesn't have a checked_add function.  atomics does though.  if you wanted to have the checked_add you would have to convert to atomics or add some custom validation on maximum value add operation